pkg/config/validation/validation.go

```diff
@@ -2347,6 +2381,30 @@ func ValidateProtocol(protocolStr string) error {
 	return nil
 }
 
+// wrapper around multierror.Append that enforces the invariant that if all input errors are nil, the output
+// error is nil (allowing validation without branching).
+func appendValidation(v Validation, vs ...error) Validation {
+	appendError := func(err, err2 error) error {
```

Comment:
nit: can we have a better naming? As it can accept both `err` and `warning`.
`err`, `err2` doesn't look good.